lms/djangoapps/bulk_email/tests/test_email.py

```diff
@@ -44,50 +46,164 @@ def mock_update_subtask_status(entry_id, current_task_id, new_subtask_status):
+
+        self.login_as_user(self.instructor)
+
+        self.goto_instructor_dash_email_view()
+        self.send_mail_url = reverse(
+            'send_email', kwargs={'course_id': unicode(self.course.id)}
+        )
         self.success_content = {
-            'course_id': self.course.id.to_deprecated_string(),
+            'course_id': unicode(self.course.id),
             'success': True,
         }
 
 
 @attr('shard_1')
 @patch.dict(settings.FEATURES, {'ENABLE_INSTRUCTOR_EMAIL': True, 'REQUIRE_COURSE_EMAIL_AUTH': False})
+class TestLocalizedFromAddress(EmailSendFromDashboardTestCase):
+
+    original_ugettext = None
+    mocked_lang = 'ar'
+
+    def setUp(self):
+        super(TestLocalizedFromAddress, self).setUp()
+
+        translations = translation.trans_real._translations  # pylint: disable=protected-access
+
+        with translation.override(self.mocked_lang):
+            # In order to undo it later
+            self.original_ugettext = translations[self.mocked_lang].ugettext
+
+            mocked_ugettext = self.get_mocked_ugettext(self.mocked_lang)
+            translations[self.mocked_lang].ugettext = mocked_ugettext
+
+    def get_mocked_ugettext(self, lang_code):
+        """
+        Mocks ugettext to return the lang code with the original string.
+
+        e.g.
+
+        >>> ugettext = self.mock_ugettext('ar')
+        >>> ugettext('Hello') == '@AR Hello@'
+        """
+        def mocked_ugettext(msg):
+            """
+            A mock of ugettext to isolate it from the real `.mo` files.
+            """
+            return u'@{} {}@'.format(lang_code.upper(), msg)
+
+        return mocked_ugettext
+
+    def send_email(self):
+        """
+        Sends a dummy email to check the `from_addr` translation.
+        """
+        test_email = {
+            'action': 'send',
+            'send_to': 'myself',
+            'subject': 'test subject for myself',
+            'message': 'test message for myself'
+        }
+
+        self.client.post(self.send_mail_url, test_email)
+
+        return mail.outbox[0]
+
+    @override_settings(LANGUAGE_CODE='en')
+    def test_english_platform(self):
+        """
+        Test if the email `from` is localized to the platform's preference.
+        """
+
+        message = self.send_email()
+
+        self.assertNotRegexpMatches(
+            message.from_email,
+            '@.* Course Staff@'
+        )
+
+    @override_settings(LANGUAGE_CODE='ar')
+    def test_arabic_platform(self):
+        """
+        Test if the email `from` is localized to the platform's preference.
+        """
+
+        message = self.send_email()
+
+        self.assertRegexpMatches(
+            message.from_email,
+            '@AR .* Course Staff@'
+        )
+
+    def tearDown(self):
+        super(TestLocalizedFromAddress, self).tearDown()
+
+        translations = translation.trans_real._translations  # pylint: disable=protected-access
```

Comment:
Oh, I don't think tearDown is bad, and I'm not saying you can't use it. I just recently saw addCleanup in a PR and thought it was a nicer paradigm because you add the cleanup code in the same place where the items that need cleaning up are defined (which means it is more likely to be accurate).